### ISSUE TITLE
Browser support policy

### ISSUE THREAD

**Issue by ChristianMayer** [CometVisu/CometVisu#1409]
Based on #1408 we should discuss and decide on a policy which browser we do support.

Generic considerations:

- Browser update themselves now, so having recent browsers is much more common now than it was a few years ago -> we could aggressively go to the newest versions
- Especially in our target market (home automation) there are devices and browsers used for displaying the CometVisu that aren't kept up to date as it is either not deemed necessary (no cybersecurity issue as only a fixed and trusted page is shown) or it might not even possible (old wall mounted PCs or tables with ancient OS)

Based on those points I'd say that we must guarantee (as far as sensibly possible) that within one release all patch versions follow the same minimum browser version that the initial release had (e.g. 0.13.1, 0.13.2, ... must follow the same minimum that 0.13.0 had set)

For a new release we could increase the minimum. Querying https://browsersl.ist/ (e.g. with https://browsersl.ist/#q=since+2022-06-26) I could come up with different sensible minimums.

- 0.13: as it's not released yet, it's not possible to know the browser versions yet
- 0.13 minus 1 year: as it's not released yet, it's not possible to know the browser versions yet
- 0.12.0 (2022-06-26): Chrome 104, Safari@iOS 15.6, Edge 104, Firefox 102
- 0.12.0 minus 1 year (2021-06-26): Chrome 92, Safari@iOS 15.0, Edge 92, Firefox 90
- 0.11.0 (2019-03-10): Chrome 73, Safari@iOS 12.2, Edge 79, Firefox 66
- 0.11.0 minus 1 year (2018-03-10): Chrome 66, Safari@iOS 11.3, Edge 17, Firefox 59

My personal opinion is that supporting the browsers that were active when the last release had happened (i.e. 0.12 for the 0.13 release) could be a sensible choice. When we want to stay more conservative last release minus one year would also be fine.

**Comment by ChristianMayer** [comment]
Updated the list with the 1.0.0 release browser versions

**Comment by joltcoke** [comment]
I agree with your proposal. However I'd propose to differentiate between: "CV doesn't support it out of the box but can be poly filled" and "not possible to get it up and running on older browsers without a code change". PR's falling into the latter group should be handled a lot more carefully.

Background: Wall installed devices probably won't get updated regularly (both hardware and software), so we have to deal with legacy installations somehow.

My current setup includes iPads running iOS 16.4 (Released 2023, latest release supported on the hardware). Doesn't work out of the box, but can be fixed with 1 or 2 poly fills. 

How about documenting the policy and also providing a list of workarounds to get older devices running? Happy to put together a proposal.

**Comment by ChristianMayer** [comment]
@joltcoke it's not so simple: the code gets translated (transpiled) with babel. Bumping the minimum versions means that there's less polyfilling required in the code.

On the other hand: when you need an old version, you can easily tune down the babel config (see e.g. #1495 about where and how) and you can create a local version that runs with the older hardware.

But as I see this question coming up more often, as we are in an environment where people are building hardware in their homes that gets outdated but still shouldn't be replaced, I could imagine to publish an "ancient hardware edition" in parallel. 
@peuter what's your opinion on that?

**Comment by joltcoke** [comment]
Agreed, having a "dual" build would solve my problem. But before considering it, what is the benefit of #1409? I guess the file size will shrink a bit. Any other advantages?

**Comment by ChristianMayer** [comment]
You can use web technologies that can't be patched. And code cleanup. That's why this here is a question and not a decision - something to find a consensus and align at

**Comment by joltcoke** [comment]
I'd say, if the only benefit is to potentially unlock something in the future, let's not do it now. From my point of view the babel settings should be a low as our codebase supports without pulling in major drawbacks because of incomplete polyfills etc.

If we encounter the option to improve something in the codebase (e.g. removing lots of custom code, upgrade to next version of library xyz) we should evaluate benefit of the codebase change vs. potential install base we loose because of the increased browser version requirements.
